pkg/messaging/pair/server.go

```diff
@@ -0,0 +1,86 @@
+package pair
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.nanomsg.org/mangos/v3/protocol"
+	"go.nanomsg.org/mangos/v3/protocol/pair"
+	"nodemon/pkg/storing/nodes"
+)
+
+func StartPairMessagingServer(ctx context.Context, nanomsgURL string, ns *nodes.Storage) error {
+	if len(nanomsgURL) == 0 || len(strings.Fields(nanomsgURL)) > 1 {
+		log.Printf("Invalid nanomsg IPC URL for pair socket'%s'", nanomsgURL)
+		return errors.New("invalid nanomsg IPC URL for pair socket")
+	}
+	socketPair, err := pair.NewSocket()
+	if err != nil {
+		log.Printf("Failed to get new pair socket: %v", err)
+		return err
+	}
+	defer func(socketPair protocol.Socket) {
+		if err := socketPair.Close(); err != nil {
+			log.Printf("Failed to close pubsub socket: %v", err)
+		}
+	}(socketPair)
+
+	if err := socketPair.Listen(nanomsgURL); err != nil {
+		log.Printf("Failed to listen on pair socket: %v", err)
+		return err
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			msg, err := socketPair.Recv()
+			if err != nil {
+				log.Printf("failed to receive a message from pair socket: %v", err)
+				return nil
+			}
+			request := RequestPairType(msg[0])
+			switch request {
+			case RequestNodeListT:
+				nodes, err := ns.Nodes()
+				if err != nil {
+					log.Printf("failed to receive list of nodes from storage, %v", err)
+				}
+
+				var nodeList NodeListResponse
+				for _, node := range nodes {
+					nodeList.Urls = append(nodeList.Urls, node.URL)
+				}
```

Comment:
You can preallocate necessary amount of memory.

Reply:
I don't think I need it here. The amount of monitored nodes is small enough, so using indices is unfounded

Reply:
Yep, It's up to you. That's just friendly recommendation.